proxygen/lib/http/codec/compress/experimental/simulator/CompressionSimulator.cpp

```diff
@@ -155,6 +160,35 @@ void CompressionSimulator::run() {
     int(100 - double(100 * stats_.compressed) / stats_.uncompressed);
 }
 
+void CompressionSimulator::flushRequests(CompressionScheme *scheme) {
+  VLOG(5) << "schedule encode for " << scheme->packetIndices.size() << " blocks at " << scheme->prev.count();
+  // Flush previous train
+  scheduleEvent([this, scheme, indices=std::move(scheme->packetIndices)] () mutable {
+      bool newPacket = true;
+      while (!indices.empty()) {
+        int16_t index = indices.front();
+        indices.pop_front();
+        auto schemeIndex = scheme->index;
+        auto encodeRes = encode(scheme, newPacket, index);
+        FrameFlags flags =encodeRes.first;
+        bool allowOOO = flags.allowOOO;
+        if (schemeIndex < minOOOThresh()) {
+          allowOOO = false;
+          auto ack = scheme->getAck(schemeIndex);
+          if (ack) {
+            scheme->recvAck(std::move(ack));
+          }
+        }
+        stats_.allowedOOO += (allowOOO) ? 1 : 0;
+        flags.allowOOO = allowOOO;
+        scheme->encodedBlocks.emplace_back(
+            flags, newPacket, std::move(encodeRes.second), &callbacks_[index]);
+        newPacket = false;
```

Comment:
Why is newPacket necessarily false here?  

The old code was trying to coalesce packets that were encoded in the same event loop.  This made things hard to reproduce (eg: turning up logging would increase coalescing, since the loop runs slower), so I like the idea of driving making the encodes all happen at the same time.

Do we want the packetization logic from flushSchemePackets in this function instead?  That way we can be sure that newPacket is always passed correctly to encode(), and it might make the code simpler by getting rid of the runInLoop concept for coalescing.

Reply:
I agree that moving packetization logic from flushSchemePackets to here would be an improvement.   

This was a step in that direction, the main thing here was the assumption that it was unlikely that partial packets would be held across non-trivial gaps in the har timeline.   So forcing new packets there is complementary to splitting at MTU boundaries.

Reply:
Do you want to take a stab at it now, or would you rather merge as is and we can make a second pass to unify that logic?

Reply:
Maybe the latter, merge as is and unify the logic in a followup.